mettle/src/crypttlv.c

```diff
@@ -41,25 +57,38 @@ struct tlv_encryption_ctx* create_tlv_encryption_context(unsigned int enc_flag)
 {
 	struct tlv_encryption_ctx *ctx = malloc(sizeof(struct tlv_encryption_ctx));
 	ctx->flag = enc_flag;
-
+	ctx->is_weak_key = false;
 #ifndef __MINGW32__
 	switch (enc_flag) {
 		case ENC_AES256: {
 			mbedtls_ctr_drbg_context ctr_drbg;
 			mbedtls_entropy_context entropy;
-
+			int error_code = 0;
 			mbedtls_entropy_init(&entropy);
 			mbedtls_ctr_drbg_init(&ctr_drbg);
-			if (!(mbedtls_ctr_drbg_seed(&ctr_drbg, mbedtls_entropy_func, &entropy, NULL, 0))) {
+
+			error_code = mbedtls_ctr_drbg_seed(&ctr_drbg, mbedtls_entropy_func, &entropy, NULL, 0);
+			if (error_code != 0) {
+				mbedtls_ctr_drbg_free(&ctr_drbg);
+				mbedtls_entropy_free(&entropy);
+				add_weak_encryption(&ctr_drbg, &entropy);
+				error_code = mbedtls_ctr_drbg_seed(&ctr_drbg, mbedtls_entropy_func, &entropy, NULL, 0);
+				if(error_code == 0) { ctx->is_weak_key = true; };
```

Comment:
When I went to test this on my Fedora system where I found the original issue, I noticed that `error_code` was not 0, which seemed to imply that this was still failing. There was also no logging which made this difficult to track down.

Would you mind please testing and landing #256 then rebasing your changes on top of it so we can get some logging moving forward? That should help track down why it's still failing.

Reply:
I analyzed the problem, the issue is that mbedtls still fails if **any** source of entropy fails, in our case `mbedtls_platform_entropy_poll`, during my tests I disabled the platform entropy so I did not encounter this problem, I modified add_mtwister_entropy_source to replace the callback of any strong source of entropy with the mtwister callback, something like this.
```c
void add_mtwister_entropy_source(void  *ctr_drbg, void *entropy) {
#ifndef __MINGW32__
	mbedtls_entropy_init((mbedtls_entropy_context *) entropy);
	mbedtls_ctr_drbg_init((mbedtls_ctr_drbg_context *)ctr_drbg);

	mbedtls_entropy_context *p_entropy = entropy;
	for(int i = 0; i < p_entropy->source_count; i++) {
		if(p_entropy->source[i].strong == MBEDTLS_ENTROPY_SOURCE_STRONG) {
			p_entropy->source[i].f_source = mbedtls_mtwister_entropy_poll;
		}

	}
#endif
}
```
It works but I'm thinking a cleaner way to do it.

Reply:
Marking this as resolved. I tested the latest changes and I see the issue is now fixed. I like the pattern of using `mbedtls_entropy_remove_source` then adding the source we want.